File: src/js/02_lengthOfLastWord.js
// leetcode
// 58. Length of Last Word

// first solution

// const lengthOfLastWord = (s) => {
//   const strArr = s.split(' ');
//   let i = strArr.length - 1;

//   while (i >= 0) {
//     const word = strArr[i];

//     if (word) return word.length;
//     i--;
//   }

//   return 0;
// };

// second solution

const lengthOfLastWord = (s) => {
  let result = '';
  let i = s.length - 1;

  while (i >= 0) {
    const letter = s[i];

    if (letter === ' ') {
      if (!result) i--;
      else break;
    } else {
      result += letter;
      i--;
    }
  }

  return result.length;
};

console.log(
  lengthOfLastWord('Hello World'),
);

File: src/py/02_lengthOfLastWord.py
# leetcode
# 58. Length of Last Word

# first solution

class Solution:
    def lengthOfLastWord(self, s: str):
        result = ''
        i = len(s) - 1

        while (i >= 0):
            if s[i] == ' ':
                if not result:
                    i -= 1
                else:
                    break
            else:
                result += s[i]
                i -= 1

        return len(result)


print(
    Solution().lengthOfLastWord('Hello World'),
    Solution().lengthOfLastWord('a')
)
